### PR TITLE
Swift3 syntax compatible with Xcode 8 beta 5

### PR DESCRIPTION
**Pull request by hyouuu**

### CODE REVIEW

**Comment by JanGorman** [comment]
Could you update the `.travis.yml` file to use `osx_image: xcode8` which should make it build there again.

File: Agrume/Agrume.swift [hunk 1]
```diff
@@ -14,20 +14,20 @@ public protocol AgrumeDataSource {
   ///
   /// - Parameter index: The index (collection view item) being displayed
   /// - Parameter completion: The completion that returns the image to be shown at the index
-	func imageForIndex(index: Int, completion: (UIImage?) -> Void)
+	func imageForIndex(_ index: Int, completion: (UIImage?) -> Void)
```
Comment (JanGorman) [inline Agrume/Agrume.swift]: I guess the Swift 3 way would be to rename the method to `fun image(forIndex index: Int, …)`

File: Agrume/Agrume.swift [hunk 2]
```diff
@@ -166,46 +166,46 @@ public final class Agrume: UIViewController {
...
-        UIView.animateWithDuration(Agrume.TransitionAnimationDuration,
+      viewController.present(self, animated: false) {
+        UIView.animate(withDuration: Agrume.TransitionAnimationDuration,
                                    delay: 0,
-                                   options: [.BeginFromCurrentState, .CurveEaseInOut],
+                                   options: .beginFromCurrentState,
                                    animations: { [weak self] in
                                       self?.collectionView.alpha = 1
-                                      self?.collectionView.transform = CGAffineTransformIdentity
+                                      self?.collectionView.transform = CGAffineTransform.identity
```
Comment (JanGorman) [inline Agrume/Agrume.swift]: could be shortened to just `self?.collectionView.transform = .identity`

File: Agrume/Agrume.swift [hunk 3]
```diff
@@ -223,13 +223,13 @@ public final class Agrume: UIViewController {
     self.dismissAfterFlick()
   }
 
-  public func showImageAtIndex(index : Int) {
-    collectionView.scrollToItemAtIndexPath(NSIndexPath(forRow: index, inSection: 0), atScrollPosition: [],
+  public func showImageAtIndex(_ index : Int) {
```
Comment (JanGorman) [inline Agrume/Agrume.swift]: `public func showImage(at index: Int)`

File: Agrume/AgrumeCell.swift [hunk 4]
```diff
@@ -189,7 +189,7 @@ extension AgrumeCell: UIGestureRecognizerDelegate {
 
     let inset: UIEdgeInsets
     if minContentWidth > contentView.bounds.width && minContentHeight > contentView.bounds.height {
-      inset = UIEdgeInsetsZero
+      inset = UIEdgeInsets.zero
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift]: `inset = .zero`

File: Agrume/AgrumeCell.swift [hunk 5]
```diff
@@ -210,25 +210,25 @@ extension AgrumeCell: UIGestureRecognizerDelegate {
...
+    let velocity = gesture.velocity(in: gesture.view)
     let vectorDistance = sqrt(pow(velocity.x, 2) + pow(velocity.y, 2))
 
-    if gesture.state == .Began {
-      isDraggingImage = CGRectContainsPoint(imageView.frame, locationInView)
+    if gesture.state == .began {
+      isDraggingImage = imageView.frame.contains(locationInView)
       if isDraggingImage {
-        startImageDragging(locationInView, translationOffset: UIOffsetZero)
+        startImageDragging(locationInView, translationOffset: UIOffset.zero)
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift]: `startImageDragging(locationInView, translationOffset: .zero)`

File: Agrume/AgrumeCell.swift [hunk 6]
```diff
@@ -268,29 +268,29 @@ extension AgrumeCell: UIGestureRecognizerDelegate {
...
 
-  private func cancelCurrentImageDrag(animated: Bool) {
+  private func cancelCurrentImageDrag(_ animated: Bool) {
     animator.removeAllBehaviors()
     attachmentBehavior = nil
     isDraggingImage = false
 
     if !animated {
-      imageView.transform = CGAffineTransformIdentity
+      imageView.transform = CGAffineTransform.identity
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift]: `imageView.transform = .identity`

File: Agrume/AgrumeCell.swift [hunk 7]
```diff
@@ -332,19 +332,19 @@ extension AgrumeCell: UIGestureRecognizerDelegate {
     }
 
     frame.size = CGSize(width: targetWidth, height: targetHeight)
-    frame.origin = CGPointZero
+    frame.origin = CGPoint.zero
```
Comment (JanGorman) [inline Agrume/AgrumeCell.swift]: `frame.origin = .zero`

**Comment by JanGorman** [comment]
Some minors but otherwise looks good 👍

**Comment by hyouuu** [comment]
Well I think cocoapod lint doesn't work with Xcode 8 yet... :(

**Comment by JanGorman** [comment]
why am I not surprised… I merged this anyway now, will take a look. Thanks for your help!

**Pull request merged**